keanu-project/src/main/java/io/improbable/keanu/algorithms/PosteriorSamplingAlgorithm.java

```diff
@@ -22,4 +23,7 @@ default NetworkSamples getPosteriorSamples(BayesianNetwork bayesianNetwork, int
         return getPosteriorSamples(bayesianNetwork, bayesianNetwork.getTopLevelLatentVertices(), sampleCount);
     }
 
+    NetworkSamplesGenerator generatePosteriorSamples(final BayesianNetwork bayesianNetwork,
+                                                     final List<? extends Vertex> verticesToSampleFrom);
+
```

Comment:
now you can give the other method a default version:

    default NetworkSamples getPosteriorSamples(BayesianNetwork bayesNet,
                                       List<? extends Vertex> verticesToSampleFrom,
                                       int sampleCount) {
        return generatePosteriorSamples(bayesNet, verticesToSampleFrom)
            .generate(sampleCount);
    }

Reply:
Good catch - will update